test_scripts/WebEngine/028_OnExitApplication_unregister.lua

```diff
@@ -0,0 +1,49 @@
+--[[ General test configuration ]]
+config.defaultMobileAdapterType = "WS"
+
+--[[ Required Shared libraries ]]
+local common = require('test_scripts/WebEngine/commonWebEngine')
+local runner = require('user_modules/script_runner')
+
+--[[ Local Variables ]]
+  local appProp = {
+    nicknames = { "Test Application" },
+    policyAppID = "0000001",
+    enabled = true,
+    authToken = "ABCD12345",
+    transportType = "WS",
+    hybridAppPreference = "CLOUD"
+  }
+
+--[[ Local Functions ]]
+local function setAppProperties()
+  common.checkUpdateAppList(appProp.policyAppID, 1, 1)
+  common.setAppProperties(appProp)
+end
+
+local function onExitApp()
+  local appID = common.getHMIAppId()
+  common.getHMIConnection():SendNotification("BasicCommunication.OnExitApplication", {reason = "RESOURCE_CONSTRAINT", appID = appID})
```

Comment:
@mvorobio Please move `BasicCommunication.OnExitApplication` notification sending after registration of all expectations.

Reply:
@aderiabin Please check https://github.com/mvorobio/sdl_atf_test_scripts/pull/1/commits/285c8094c242f73fd09d3a3d348e61f456a7e233